src/python/WMCore/WMRuntime/Tools/Scram.py

```diff
@@ -110,7 +110,9 @@ def preCommand(self):
 
         """
         result = ""
-        if self.architecture != None:
+        if self.architecture == None:
```

Comment:
BTW, why you explicitly export SCRAM_ARCH (without value) now. Is it really needed?